### PR TITLE
C backend: use static_cast<>/reinterpret_cast<> instead of C-style casts

### PR DESCRIPTION
**Pull request by steven-johnson**
Another stylistic change that shouldn't affect object code at all, but IMHO makes the C code a bit easier to read and reason about. (Disclaimer: not sure if this will affect the GPU, etc backends that share this code -- do Cuda et al support this syntax?)

### CODE REVIEW

File: src/CodeGen_C.cpp [hunk 1]
```diff
@@ -1375,14 +1375,14 @@ string CodeGen_C::print_type(Type type, AppendSpaceIfNeeded space_option) {
...
     if (type.is_handle() || e.type().is_handle()) {
-        // Use a c-style cast if either src or dest is a handle --
+        // Use a reinterpret_cast<> if either src or dest is a handle --
         // note that although Halide declares a "Handle" to always be 64 bits,
         // the source "handle" might actually be a 32-bit pointer (from
-        // a function parameter), so calling reinterpret<> (which just memcpy's)
+        // a function parameter), so calling halide_reinterpret_bits<> (which just memcpy's)
         // would be garbage-producing.
-        oss << "(" << print_type(type) << ")";
+        oss << "reinterpret_cast<" << print_type(type) << ">";
```
Comment (abadams) [inline src/CodeGen_C.cpp:1383]: I don't think a reinterpret cast is valid here, because the bit-width might change.

**Comment by abadams** [comment]
Doesn't this just make the generated code more verbose? We don't tend to use static_cast inside libHalide so I'm not sure why we'd prefer to generate it.

**Comment by steven-johnson** [comment]
> Doesn't this just make the generated code more verbose? We don't tend to use static_cast inside libHalide so I'm not sure why we'd prefer to generate it.

My theory here was that when generating code, we can select the right cast more appropriately. (Note that this was opened as a draft...)

**Comment by steven-johnson** [comment]
This breaks various GPU backends, and is definitely not worth conditionalizing for that -- closing as a failed experiment.

**Pull request closed without merging**